contracts/providers/chainlink/ChainlinkPairAggregator.sol

```diff
@@ -38,15 +53,31 @@ contract ChainlinkPairAggregator is PairAggregatorInterface {
         return aggregator.latestRound();
     }
 
-    function getPreviousAnswer(uint256 roundsBack) external view returns (int256) {
+    function getPreviousTimestamp(uint256 roundsBack) external view returns (uint256) {
         uint256 latest = aggregator.latestRound();
-        require(roundsBack <= latest, "Not enough history");
-        return aggregator.getAnswer(latest - roundsBack);
+        require(roundsBack <= latest, "NOT_ENOUGH_HISTORY");
+        return aggregator.getTimestamp(latest - roundsBack);
     }
 
-    function getPreviousTimestamp(uint256 roundsBack) external view returns (uint256) {
+    /** Internal Functions */
+
+    function _getPreviousAnswer(uint256 roundsBack) internal view returns (int256) {
         uint256 latest = aggregator.latestRound();
-        require(roundsBack <= latest, "Not enough history");
-        return aggregator.getTimestamp(latest - roundsBack);
+        require(roundsBack <= latest, "NOT_ENOUGH_HISTORY");
+        return aggregator.getAnswer(latest - roundsBack);
+    }
+
+    function _normalizeDecimals(int256 value) internal view returns (int256) {
+        if( collateralDecimals >= responseDecimals) {
+            uint8 pendingDecimals = collateralDecimals - responseDecimals;
+            return value * int256(TEN ** pendingDecimals);
+        } else {
+            uint8 pendingDecimals = responseDecimals - collateralDecimals;
+            return value / int256(TEN ** pendingDecimals);
+        }
+    }
+
+    function _normalizeResponse(int256 response) internal view returns (int256) {
```

Comment:
What's the purpose of this function? Why not just call `_normalizeDecimals` directly instead?

Reply:
I see its so it works with the inverseaggregator...
why not just define `_normalizeResponse` as the above decimal work, and then in the inverse aggregator defined `_normalizeResponse` as the inversion maths? It seems weird to just have a few function calls that purely call another function

Reply:
Okay, so I removed the _normalizeDecimals function (moved logic to _normalizeResponse). The inverse aggregator already has the _normalizeResponse as you mentioned.